### PR TITLE
[PRODDEV-179] Store references entities' info in the metadata field, …

### PR DESCRIPTION
**Pull request by NightHunterSV**
…and use this info if the entity is not there already

https://openy.atlassian.net/browse/PRODDEV-179

## Steps to test:

- [ ] Log in as an admin
- [ ] Enable "Open Y Virtual YMCA Log" module
- [ ] Navigate through the site
- [ ] In the incognito browser mode or another browser log in as a common VY user
- [ ] Navigate through the site again
- [ ] Ensure the logs are displayed on the /admin/virtual-y-logs page
image

- [ ] Export the logs
- [ ] Ensure all the info is there
- [ ] Remove the dummy user you've used to navigate through the VY
- [ ] Ensure the logs are still available -- just the user's email is no more clickable
image

- [ ] Export the logs and ensure the resulting file coincides with the previously exported
- [ ] Remove some of the entities you've looked, when surfed around the VY
- [ ] Ensure the logs are still available and contain the entities titles and instructor name of the removed entity
- [ ] Export the logs and ensure the resulting file coincides with the previously exported

## Quality checks:

Please check these boxes to confirm this PR covers the following cases:

- [x] This PR  provides updates to support the upgrade path
- [x] This change does not contain front-end fixes.
- [x] I have flagged this PR "Needs Review" or pinged the VY devs/QA
  team in Slack

### CODE REVIEW

File: modules/openy_gc_log/openy_gc_log.install [hunk 1]
```diff
@@ -131,3 +131,28 @@ function openy_gc_log_update_8006() {
...
+function openy_gc_log_update_8007() {
+  $config_dir = drupal_get_path('module', 'openy_gc_log');
+  $config_dir .= '/config/install/';
+  $config_importer = \Drupal::service('openy_upgrade_tool.importer');
+  $config_importer->setDirectory($config_dir);
+  $config_importer->importConfigs([
+    'views.view.virtual_y_logs',
+  ]);
+
+  // Update entity to add the changed field.
```
Comment (anpolimus) [inline modules/openy_gc_log/openy_gc_log.install:147]: I think it is better to separate views update and field install.
Not required, but if you have time, better for reading.

File: modules/openy_gc_log/src/Entity/LogEntity.php [hunk 2]
```diff
@@ -268,6 +269,20 @@ public static function defineChangedField(array &$fields) {
...
+   * Define event_metadata field.
+   *
+   * @param array $fields
+   *   Fields.
+   */
+  public static function defineMetadataField(array &$fields) {
+    $fields['event_metadata'] = BaseFieldDefinition::create('string')
+      ->setLabel(t('Event Metadata'))
+      ->setDescription(t('The event metadata, related to the Log entity.'))
+      ->setSetting('max_length', 255)
```
Comment (anpolimus) [inline modules/openy_gc_log/src/Entity/LogEntity.php:282]: let's make it larger in case if we need to put more info to the meta in the future.

**Review by anpolimus — CHANGES_REQUESTED** [review]

**Review by anpolimus — APPROVED** [review]

**Comment by AnastasiiaPys** [comment]
Tested, all good now!

**Review by hamrant — APPROVED** [review]

File: modules/openy_gc_log/config/install/views.view.virtual_y_logs.yml [hunk 3]
```diff
@@ -928,7 +1058,18 @@ display:
...
+        area_text_custom:
+          id: area_text_custom
+          table: views
+          field: area_text_custom
+          relationship: none
+          group_type: group
+          admin_label: ''
+          empty: true
+          tokenize: false
+          content: 'Below are this month''s logs. Older log entries can be found in the <a href=/admin/content/files?filename=virtual-y&filemime=application/x-gzip>Log Archives</a>.'
```
Comment (anpolimus) [inline modules/openy_gc_log/config/install/views.view.virtual_y_logs.yml:1071]:
```suggestion
          content: '<div class="messages messages--warning">Below are this month's logs. Older log entries can be found in the <a href=/admin/content/files?filename=virtual-y&filemime=application/x-gzip>Log Archives</a>.</div>'
```

File: modules/openy_gc_log/config/install/views.view.virtual_y_logs.yml [hunk 4]
```diff
@@ -928,7 +1058,18 @@ display:
...
+        area_text_custom:
+          id: area_text_custom
+          table: views
+          field: area_text_custom
+          relationship: none
+          group_type: group
+          admin_label: ''
+          empty: true
+          tokenize: false
+          content: '<div class="messages messages--warning">Below are this month's logs. Older log entries can be found in the <a href=/admin/content/files?filename=virtual-y&filemime=application/x-gzip>Log Archives</a>.</div>'
```
Comment (anpolimus) [inline modules/openy_gc_log/config/install/views.view.virtual_y_logs.yml:1071]:
```suggestion
          content: '<div class="messages messages--warning">Below are this month\'s logs. Older log entries can be found in the <a href=/admin/content/files?filename=virtual-y&filemime=application/x-gzip>Log Archives</a>.</div>'
```

File: modules/openy_gc_log/config/install/views.view.virtual_y_logs.yml [hunk 5]
```diff
@@ -928,7 +1058,18 @@ display:
...
+        area_text_custom:
+          id: area_text_custom
+          table: views
+          field: area_text_custom
+          relationship: none
+          group_type: group
+          admin_label: ''
+          empty: true
+          tokenize: false
+          content: '<div class="messages messages--warning">Below are this month\'s logs. Older log entries can be found in the <a href=/admin/content/files?filename=virtual-y&filemime=application/x-gzip>Log Archives</a>.</div>'
```
Comment (anpolimus) [inline modules/openy_gc_log/config/install/views.view.virtual_y_logs.yml:1071]:
```suggestion
          content: '<div class="messages messages--warning">Below are this months logs. Older log entries can be found in the <a href=/admin/content/files?filename=virtual-y&filemime=application/x-gzip>Log Archives</a>.</div>'
```

**Pull request merged**